### PR TITLE
[ROCM] remove rocfft workaround

### PR DESCRIPTION
**Pull request by micmelesse**
We enable the last remaining fft tests on rocm. The issue was caused by a workaround which is no longer needed.

### CODE REVIEW

**Review by pruthvistony — APPROVED** [review]

**Comment by pruthvistony** [comment]
@micmelesse ,
Can you please check if the failure is happening due to your PR.
```
The PR is introducing backward incompatible changes to the operator library. Please contact PyTorch team to confirm whether this change is wanted or not. 

Broken ops: [
	aten::expand.SymInt(Tensor(a) self, SymInt[] size, *, bool implicit=False) -> (Tensor(a))
	aten::chalf(Tensor self, *, int? memory_format=None) -> (Tensor)
	aten::expand_copy.SymInt(Tensor self, SymInt[] size, *, bool implicit=False) -> (Tensor)
]
```
Since it is only a test related change, not sure about the above error.
Can you also try re-triggering a CI run.

**Comment by micmelesse** [comment]
@pruthvistony Everything passes. I just had to merge in main

File: test/test_spectral_ops.py [hunk 1]
```diff
@@ -566,11 +467,7 @@ def fn(t: torch.Tensor, s: Optional[List[int]], dim: List[int] = (-2, -1), norm:
 
                 torch_fns = (torch_fn, torch.jit.script(fn))
 
-                if torch.version.hip is not None:
-                    valid_input_default, s, _, norm = self._generate_valid_rocfft_input(
-                        input, torch_fn, s, None, norm)
-                else:
-                    valid_input_default = input
+                valid_input_default = input
```
Comment (albanD) [inline test/test_spectral_ops.py:470]: Can we just use input throughout the test now and remove both `valid_input_default` and `valid_input_explicit` which are the same thing?

**Review by albanD — COMMENTED** [review]
Thanks for the cleanup!
Quick question about simplifying the test to make it easier to read.

File: test/test_spectral_ops.py (hunk 1 above)
Comment (micmelesse) [reply test/test_spectral_ops.py]: Done

**Comment by micmelesse** [comment]
@albanD @pruthvistony I have addressed the feedback.

**Review by jeffdaily — APPROVED** [review]
LGTM.  CI passing.

**Review by albanD — APPROVED** [review]
Thanks!

**Pull request closed without merging**